content/en/docs/concepts/overview/object-management-kubectl/kustomization.md

```diff
@@ -0,0 +1,752 @@
+kustomization directories as its `bases`. A **base** has no knowledge of an overlay and can be used in multiple overlay.
+An overlay may have multiple bases and it composes all resources
+from bases and may also have customization on top of them.
+
+Here is an example of a base.
+```shell
+# Create a directory to hold the base
+mkdir base
+# Create a base/deployment.yaml
+at <<EOF > base/deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-nginx
+spec:
+  selector:
+    matchLabels:
+      run: my-nginx
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        run: my-nginx
+    spec:
+      containers:
+      - name: my-nginx
+        image: nginx
+        command: ["start", "--host", "\$(MY_SERVICE_NAME)"]
+EOF
+
+# Create a base/service.yaml file
+cat <<EOF > base/service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-nginx
+  labels:
+    run: my-nginx
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+  selector:
+    run: my-nginx
+EOF
+# Create a base/kustomization.yaml
+cat <<EOF > base/kustomization.yaml
+resources:
+- deployment.yaml
+- service.yaml
+```
+This base can be used in multiple overlays. You can add different namePrefix or other cross-cutting fields
+in different overlays. Here are two overlays using the same base.
+```shell
+mkdir dev
+cat <<EOF > dev/kustomization.yaml
+bases:
+- ../base
+namePrefix: dev-
+EOF
+
+mkdir prod
+cat <<EOF > prod/kustomization.yaml
+bases:
+- ../base
+namePrefix: prod-
+EOF
+```
+
+## How to apply/view/delete objects from Kustomization
+Use `--kustomize` or `-k` in kubectl commands to recognize Resources managed by `kustomization.yaml`. 
+Note that `-k` should point to a kustomization directory, such as
+
+```shell
+kubectl apply -k <kustomization directory>/
+```
+Given following `kustomization.yaml` file and associated Resource files,
+```shell
+# Create a deployment.yaml file
+cat <<EOF > deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-nginx
+spec:
+  selector:
+    matchLabels:
+      run: my-nginx
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        run: my-nginx
+    spec:
+      containers:
+      - name: my-nginx
+        image: nginx
+        ports:
+        - containerPort: 80
+EOF
+
+# Create a kustomization.yaml
+cat <<EOF >./kustomization.yaml
+namePrefix: dev-
+commonLabels:
+  app: my-nginx
+resources:
+- deployment.yaml
+EOF
+```
+Running following command will apply the Deployment object `dev-my-nginx`
+```shell
+> kubectl apply -k ./
+deployment.apps/dev-my-nginx created
+```
+Run following command to view the Deployment object
+```shell
+kubectl get -k ./
+```
+or 
+```shell
+kubectl describe -k ./
+```
+Run following command to delete the Deployment object
+```shell
+> kubectl delete -k ./
+deployment.apps "dev-my-nginx" deleted
+```
+
+
+## Kustomize Feature List
+Here is a list of all the features in Kustomize.
+
```

Comment:
nit: there may be extra spaces in the table. This table renders correctly for me:

| Field                 | Type             | Explanation                                                    |
|-----------------------|------------------|----------------------------------------------------------------|
| namespace             | string           | add namespace to all resources                                 |
| namePrefix            | string           | value of this field is prepended to the names of all resources |
| nameSuffix            | string           | value of this field is appended to the names of all resources  |
| commonlabels          | map[string]string | labels to add to all resources and selectors                  |
| commonAnnotations     | map[string]string | annotations to add to all resources                           |
| resources             | []string          | each entry in this list must resolve to an existing resource configuration file |
| configmapGenerator    | [][ConfigMapArgs](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/types/kustomization.go#L195)  | Each entry in this list generates a ConfigMap |
| secretGenerator       | [][SecretArgs](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/types/kustomization.go#L201)     | Each entry in this list generates a Secret                                         |
| generatorOptions      | [GeneratorOptions](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/types/kustomization.go#L239) | Modify behaviors of all ConfigMap and Secret generators |
| bases                 | []string                                                                                                     | Each entry in this list should resolve to a directory containing a kustomization.yaml file |
| patchesStrategicMerge | []string                                                                                                     | Each entry in this list should resolve a strategic merge patch of a Kubernetes object |
| patchesJson6902       | [][Json6902](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/patch/json6902.go#L23)             | Each entry in this list should resolve to a Kubernetes object and a Json Patch     |
| vars                  | [][Var](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/types/var.go#L31)                       | Each entry is to capture text from one resource's field                            |
| images                | [][Image](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/image/image.go#L23)                   | Each entry is to modify the name, tags and/or digest for one image without creating patches |
| configurations        | []string                                                                                                     | Each entry in this list should resolve to a file containing [Kustomize transformer configurations](https://github.com/kubernetes-sigs/kustomize/tree/master/examples/transformerconfigs) |
| crds                  | []string                                                                                                     | Each entry in this list should resolve to an OpenAPI definition file for Kubernetes types |